Adopt even more smart pointers in the DOM

cdumez opened the pull request:
#### 100c20fadc5e59baa58f342cec29e7b9dff3c1e5
<pre>
Adopt even more smart pointers in the DOM
<a href="https://bugs.webkit.org/show_bug.cgi?id=263171">https://bugs.webkit.org/show_bug.cgi?id=263171</a>

Reviewed by Darin Adler and Ryosuke Niwa.

* Source/WTF/wtf/Ref.h:
Remove dead declarations (without implementation) which caused linking errors
as I was working on this.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::focusedImageMapUIElement):
(WebCore::AXObjectCache::addRelations):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::labelsForNode):
(WebCore::AccessibilityNodeObject::popoverTargetElement const):
(WebCore::AccessibilityNodeObject::textUnderElement const):
(WebCore::AccessibilityNodeObject::descriptionForElements const):
(WebCore::AccessibilityNodeObject::ariaLabeledByElements const):
(WebCore:: const): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::isTabItemSelected const):
(WebCore::AccessibilityObject::isExpanded const):
(WebCore::AccessibilityObject::elementsFromAttribute const):
(WebCore:: const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::popoverTargetElement const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::accessibilityParentForImageMap const):
(WebCore::AccessibilityRenderObject::documentLinks):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
* Source/WebCore/css/StyleSheetList.cpp:
(WebCore::StyleSheetList::namedItem const):
* Source/WebCore/dom/CustomElementDefaultARIA.cpp:
(WebCore::CustomElementDefaultARIA::elementForAttribute const):
* Source/WebCore/dom/CustomElementDefaultARIA.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::handlePopoverLightDismiss):
* Source/WebCore/dom/DocumentFragment.cpp:
(WebCore::DocumentFragment::getElementById const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::getElementAttribute const):
(WebCore::Element::getElementsArrayAttribute const):
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementInternals.cpp:
(WebCore::ElementInternals::getElementAttribute const):
* Source/WebCore/dom/ElementInternals.h:
* Source/WebCore/dom/ImageOverlay.cpp:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::renderRect):
* Source/WebCore/dom/SelectorQuery.cpp:
(WebCore::SelectorDataList::executeFastPathForIdSelector const):
(WebCore::filterRootById):
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::getElementById const):
(WebCore::TreeScope::getAllElementsById const):
(WebCore::TreeScope::getElementByName const):
(WebCore::TreeScope::getImageMap const):
(WebCore::TreeScope::imageElementByUsemap const):
(WebCore::TreeScope::labelElementsForId):
(WebCore::TreeScope::findAnchor):
(WebCore:: const): Deleted.
(): Deleted.
* Source/WebCore/dom/TreeScope.h:
(WebCore::TreeScope::hasElementWithId const): Deleted.
(WebCore::TreeScope::containsMultipleElementsWithId const): Deleted.
(WebCore::TreeScope::hasElementWithName const): Deleted.
(WebCore::TreeScope::containsMultipleElementsWithName const): Deleted.
* Source/WebCore/dom/TreeScopeInlines.h: Added.
(WebCore::TreeScope::hasElementWithId const):
(WebCore::TreeScope::containsMultipleElementsWithId const):
(WebCore::TreeScope::hasElementWithName const):
(WebCore::TreeScope::containsMultipleElementsWithName const):
* Source/WebCore/dom/TreeScopeOrderedMap.cpp:
(WebCore::TreeScopeOrderedMap::get const):
(WebCore::TreeScopeOrderedMap::getAll const):
(WebCore::TreeScopeOrderedMap::getElementById const):
(WebCore::TreeScopeOrderedMap::getElementByName const):
(WebCore::TreeScopeOrderedMap::getElementByMapName const):
(WebCore::TreeScopeOrderedMap::getElementByUsemap const):
(WebCore::TreeScopeOrderedMap::getElementsByLabelForAttribute const):
(WebCore::TreeScopeOrderedMap::getElementByWindowNamedItem const):
(WebCore::TreeScopeOrderedMap::getElementByDocumentNamedItem const):
(WebCore::TreeScopeOrderedMap::getAllElementsById const):
(WebCore:: const): Deleted.
* Source/WebCore/dom/TreeScopeOrderedMap.h:
* Source/WebCore/dom/TreeWalker.cpp:
* Source/WebCore/dom/UserGestureIndicator.cpp:
(WebCore::UserGestureToken::UserGestureToken):
* Source/WebCore/dom/VisitedLinkState.cpp:
(WebCore::VisitedLinkState::invalidateStyleForAllLinks):
(WebCore::VisitedLinkState::invalidateStyleForLink):
(WebCore::VisitedLinkState::determineLinkStateSlowCase):
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
* Source/WebCore/html/CachedHTMLCollectionInlines.h:
(WebCore::traversalType&gt;::namedItem const):
* Source/WebCore/html/HTMLAreaElement.cpp:
(WebCore::HTMLAreaElement::imageElement const):
(WebCore::HTMLAreaElement::isFocusable const):
(WebCore::HTMLAreaElement::setFocus):
* Source/WebCore/html/HTMLAreaElement.h:
* Source/WebCore/html/HTMLCollection.cpp:
(WebCore::HTMLCollection::namedItemSlow const):
(WebCore::HTMLCollection::namedItems const):
* Source/WebCore/html/HTMLCollection.h:
* Source/WebCore/html/HTMLCollectionInlines.h:
(WebCore::CollectionNamedElementCache::findElementsWithId const):
(WebCore::CollectionNamedElementCache::findElementsWithName const):
(WebCore::CollectionNamedElementCache::find const):
(WebCore::CollectionNamedElementCache::append):
(WebCore:: const): Deleted.
* Source/WebCore/html/HTMLDocument.h:
(WebCore::HTMLDocument::documentNamedItem const):
(WebCore::HTMLDocument::windowNamedItem const):
* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::popoverTargetElement const):
* Source/WebCore/html/HTMLFormControlElement.h:
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::associatedMapElement const):
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::imageElement):
* Source/WebCore/html/HTMLMapElement.h:
* Source/WebCore/inspector/InspectorAuditAccessibilityObject.cpp:
(WebCore::InspectorAuditAccessibilityObject::getControlledNodes):
(WebCore::InspectorAuditAccessibilityObject::getFlowedNodes):
(WebCore::InspectorAuditAccessibilityObject::getOwnedNodes):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForAccessibilityProperties):
* Source/WebCore/page/FocusController.cpp:
(WebCore::invokerForPopoverShowingState):
(WebCore::FocusNavigationScope::scopeOwnedByPopoverInvoker):
(WebCore::FocusController::advanceFocusDirectionally):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::FocusCandidate::FocusCandidate):
* Source/WebCore/rendering/CSSFilter.cpp:
(WebCore::referenceFilterElement):
(WebCore::isIdentityReferenceFilter):
(WebCore::calculateReferenceFilterOutsets):
(WebCore::createReferenceFilter):
* Source/WebCore/rendering/ReferencedSVGResources.cpp:
(WebCore::ReferencedSVGResources::removeClientForTarget):
(WebCore::ReferencedSVGResources::updateReferencedResources):
(WebCore::ReferencedSVGResources::elementForResourceID):
(WebCore::ReferencedSVGResources::referencedFilterElement):
* Source/WebCore/rendering/ReferencedSVGResources.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::hitTestClipPath const):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::imageMap const):
(WebCore::RenderImage::nodeAtPoint):
* Source/WebCore/rendering/RenderImage.h:
* Source/WebCore/rendering/RenderLayerFilters.cpp:
(WebCore::RenderLayerFilters::updateReferenceFilterClients):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::getElementById):
* Source/WebCore/svg/SVGViewSpec.cpp:
(WebCore::SVGViewSpec::viewTarget const):
* Source/WebCore/svg/SVGViewSpec.h:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::eventBaseFor):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* Source/WebCore/xml/XPathFunctions.cpp:
(WebCore::XPath::FunId::evaluate const):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::pageNumberForElementById):

Canonical link: <a href="https://commits.webkit.org/269380@main">https://commits.webkit.org/269380@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2996c8c405b3885caaba3d986be592301a7b9f6a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22401 "1 style error") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/15 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23480 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24305 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20734 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22655 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/20 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22934 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22640 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/20 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/19447 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25159 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/20 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20302 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/26552 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/19493 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/20 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20540 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24407 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/21772 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17866 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/25820 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/10 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6066 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5331 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/19 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/27097 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [❌ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5894 "Found 2 new JSC stress test failures: microbenchmarks/array-from-object-func.js.lockdown, microbenchmarks/array-from-object.js.lockdown (failure)") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/accessibility/AccessibilityNodeObject.h:
```diff
@@ -198,8 +198,8 @@ class AccessibilityNodeObject : public AccessibilityObject {
-    Vector<Element*> ariaLabeledByElements() const;
-    String descriptionForElements(Vector<Element*>&&) const;
+    Vector<Ref<Element>> ariaLabeledByElements() const;
+    String descriptionForElements(Vector<Ref<Element>>&&) const;
```
I wonder why this needs to adopt a vector; it’s serializing the elements, not storing references to them. I’d think this would be `const Vector<Ref<Element>>&` or `Span<const Ref<Element>>`.

darinadler commented on Source/WebCore/dom/TreeScope.cpp:
```diff
@@ -456,8 +457,8 @@ Element* TreeScope::findAnchor(StringView name)
-    if (Element* element = getElementById(name))
-        return element;
+    if (RefPtr element = getElementById(name))
+        return element.get();
```
I would have used `auto` here, not `RefPtr`.

But also, this doesn’t look safe. Why can we just return a raw pointer and let the `RefPtr` fall out of scope? That seems intrinsically dangerous. Seems like `findAnchor` needs to return a `RefPtr<Element>`.

cdumez replied:
For things that return a RefPtr / Ref, I much prefer using `RefPtr` / `Ref` type for the local variable than `auto`. It makes it much clearer that the new code is safe and it makes the effort to adopt smart pointers easier. Prior to C++17, where we had to specify `RefPtr<TYPE>` then it would have been annoying. However, now that we can just use `RefPtr / Ref` without specifying the underlying type, I don't see any big drawback to using them explicitly.

In terms of safety, I believe the code is safe. getElementById() is not creating an element, it is returning an existing one. It happens to return a RefPtr<> because it was holding a RefPtr internally anyway so there was no cost to returning a RefPtr. This also reduces ref counting churn as call sites will often store the result in a RefPtr anyway.

We may want findAnchor() to return a RefPtr still. I will look into it. As I remember it, one of the return sites didn't already have a RefPtr to return so it would have created new ref counting churn to change the return type. My policy has been to only update the return type to be a RefPtr if there is no extra cost to do so (i.e. all return sites already had a RefPtr). Otherwise, our smart pointer adoption policy doesn't require changing any return types or argument types AFAIK.

darinadler replied:
As a style question, I find some of this contradictory. We write `RefPtr` to reassure people that it’s a protected reference, but in this context that protection does no good! Why is it a `RefPtr`? The real reason is that it’s because the function result type is a `RefPtr`! So, yes, I can tell the local is “safe” by seeing the `RefPtr` but in this context that’s bogus non-information; it’s a needless `RefPtr`. Using `auto` would be an improvement in _this_ case, but, yes, it would make things worse when you need reassurance that it’s a `RefPtr`. I’m not saying any of this is wrong exactly, just that it feels random.

darinadler commented on Source/WebCore/dom/TreeScope.h:
```diff
@@ -27,7 +27,6 @@
 #pragma once
 
 #include "ExceptionOr.h"
-#include "TreeScopeOrderedMap.h"
```
Great!

cdumez committed:
Adopt even more smart pointers in the DOM
https://bugs.webkit.org/show_bug.cgi?id=263171

Reviewed by Darin Adler and Ryosuke Niwa.

* Source/WTF/wtf/Ref.h:
Remove dead declarations (without implementation) which caused linking errors
as I was working on this.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::focusedImageMapUIElement):
(WebCore::AXObjectCache::addRelations):
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::labelsForNode):
(WebCore::AccessibilityNodeObject::popoverTargetElement const):
(WebCore::AccessibilityNodeObject::textUnderElement const):
(WebCore::AccessibilityNodeObject::descriptionForElements const):
(WebCore::AccessibilityNodeObject::ariaLabeledByElements const):
(WebCore:: const): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::isTabItemSelected const):
(WebCore::AccessibilityObject::isExpanded const):
(WebCore::AccessibilityObject::elementsFromAttribute const):
(WebCore:: const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::popoverTargetElement const):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::accessibilityParentForImageMap const):
(WebCore::AccessibilityRenderObject::documentLinks):
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
(WebCore::jsDOMWindowPropertiesGetOwnPropertySlotNamedItemGetter):
* Source/WebCore/css/StyleSheetList.cpp:
(WebCore::StyleSheetList::namedItem const):
* Source/WebCore/dom/CustomElementDefaultARIA.cpp:
(WebCore::CustomElementDefaultARIA::elementForAttribute const):
* Source/WebCore/dom/CustomElementDefaultARIA.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::handlePopoverLightDismiss):
* Source/WebCore/dom/DocumentFragment.cpp:
(WebCore::DocumentFragment::getElementById const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::getElementAttribute const):
(WebCore::Element::getElementsArrayAttribute const):
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementInternals.cpp:
(WebCore::ElementInternals::getElementAttribute const):
* Source/WebCore/dom/ElementInternals.h:
* Source/WebCore/dom/ImageOverlay.cpp:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::renderRect):
* Source/WebCore/dom/SelectorQuery.cpp:
(WebCore::SelectorDataList::executeFastPathForIdSelector const):
(WebCore::filterRootById):
* Source/WebCore/dom/TreeScope.cpp:
(WebCore::TreeScope::getElementById const):
(WebCore::TreeScope::getAllElementsById const):
(WebCore::TreeScope::getElementByName const):
(WebCore::TreeScope::getImageMap const):
(WebCore::TreeScope::imageElementByUsemap const):
(WebCore::TreeScope::labelElementsForId):
(WebCore::TreeScope::findAnchor):
(WebCore:: const): Deleted.
(): Deleted.
* Source/WebCore/dom/TreeScope.h:
(WebCore::TreeScope::hasElementWithId const): Deleted.
(WebCore::TreeScope::containsMultipleElementsWithId const): Deleted.
(WebCore::TreeScope::hasElementWithName const): Deleted.
(WebCore::TreeScope::containsMultipleElementsWithName const): Deleted.
* Source/WebCore/dom/TreeScopeInlines.h: Added.
(WebCore::TreeScope::hasElementWithId const):
(WebCore::TreeScope::containsMultipleElementsWithId const):
(WebCore::TreeScope::hasElementWithName const):
(WebCore::TreeScope::containsMultipleElementsWithName const):
* Source/WebCore/dom/TreeScopeOrderedMap.cpp:
(WebCore::TreeScopeOrderedMap::get const):
(WebCore::TreeScopeOrderedMap::getAll const):
(WebCore::TreeScopeOrderedMap::getElementById const):
(WebCore::TreeScopeOrderedMap::getElementByName const):
(WebCore::TreeScopeOrderedMap::getElementByMapName const):
(WebCore::TreeScopeOrderedMap::getElementByUsemap const):
(WebCore::TreeScopeOrderedMap::getElementsByLabelForAttribute const):
(WebCore::TreeScopeOrderedMap::getElementByWindowNamedItem const):
(WebCore::TreeScopeOrderedMap::getElementByDocumentNamedItem const):
(WebCore::TreeScopeOrderedMap::getAllElementsById const):
(WebCore:: const): Deleted.
* Source/WebCore/dom/TreeScopeOrderedMap.h:
* Source/WebCore/dom/TreeWalker.cpp:
* Source/WebCore/dom/UserGestureIndicator.cpp:
(WebCore::UserGestureToken::UserGestureToken):
* Source/WebCore/dom/VisitedLinkState.cpp:
(WebCore::VisitedLinkState::invalidateStyleForAllLinks):
(WebCore::VisitedLinkState::invalidateStyleForLink):
(WebCore::VisitedLinkState::determineLinkStateSlowCase):
* Source/WebCore/dom/mac/ImageControlsMac.cpp:
* Source/WebCore/html/CachedHTMLCollectionInlines.h:
(WebCore::traversalType>::namedItem const):
* Source/WebCore/html/HTMLAreaElement.cpp:
(WebCore::HTMLAreaElement::imageElement const):
(WebCore::HTMLAreaElement::isFocusable const):
(WebCore::HTMLAreaElement::setFocus):
* Source/WebCore/html/HTMLAreaElement.h:
* Source/WebCore/html/HTMLCollection.cpp:
(WebCore::HTMLCollection::namedItemSlow const):
(WebCore::HTMLCollection::namedItems const):
* Source/WebCore/html/HTMLCollection.h:
* Source/WebCore/html/HTMLCollectionInlines.h:
(WebCore::CollectionNamedElementCache::findElementsWithId const):
(WebCore::CollectionNamedElementCache::findElementsWithName const):
(WebCore::CollectionNamedElementCache::find const):
(WebCore::CollectionNamedElementCache::append):
(WebCore:: const): Deleted.
* Source/WebCore/html/HTMLDocument.h:
(WebCore::HTMLDocument::documentNamedItem const):
(WebCore::HTMLDocument::windowNamedItem const):
* Source/WebCore/html/HTMLFormControlElement.cpp:
(WebCore::HTMLFormControlElement::popoverTargetElement const):
* Source/WebCore/html/HTMLFormControlElement.h:
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::associatedMapElement const):
* Source/WebCore/html/HTMLImageElement.h:
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::imageElement):
* Source/WebCore/html/HTMLMapElement.h:
* Source/WebCore/inspector/InspectorAuditAccessibilityObject.cpp:
(WebCore::InspectorAuditAccessibilityObject::getControlledNodes):
(WebCore::InspectorAuditAccessibilityObject::getFlowedNodes):
(WebCore::InspectorAuditAccessibilityObject::getOwnedNodes):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForAccessibilityProperties):
* Source/WebCore/page/FocusController.cpp:
(WebCore::invokerForPopoverShowingState):
(WebCore::FocusNavigationScope::scopeOwnedByPopoverInvoker):
(WebCore::FocusController::advanceFocusDirectionally):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::FocusCandidate::FocusCandidate):
* Source/WebCore/rendering/CSSFilter.cpp:
(WebCore::referenceFilterElement):
(WebCore::isIdentityReferenceFilter):
(WebCore::calculateReferenceFilterOutsets):
(WebCore::createReferenceFilter):
* Source/WebCore/rendering/ReferencedSVGResources.cpp:
(WebCore::ReferencedSVGResources::removeClientForTarget):
(WebCore::ReferencedSVGResources::updateReferencedResources):
(WebCore::ReferencedSVGResources::elementForResourceID):
(WebCore::ReferencedSVGResources::referencedFilterElement):
* Source/WebCore/rendering/ReferencedSVGResources.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::hitTestClipPath const):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::imageMap const):
(WebCore::RenderImage::nodeAtPoint):
* Source/WebCore/rendering/RenderImage.h:
* Source/WebCore/rendering/RenderLayerFilters.cpp:
(WebCore::RenderLayerFilters::updateReferenceFilterClients):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::getElementById):
* Source/WebCore/svg/SVGViewSpec.cpp:
(WebCore::SVGViewSpec::viewTarget const):
* Source/WebCore/svg/SVGViewSpec.h:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::eventBaseFor):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* Source/WebCore/xml/XPathFunctions.cpp:
(WebCore::XPath::FunId::evaluate const):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::pageNumberForElementById):

Canonical link: https://commits.webkit.org/269380@main
